File: Rust/Masters/src/models/order.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderItem {
    pub item_id: String,
    pub quantity: u32,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderRequest {
    pub items: Vec<OrderItem>,
    pub coupon_code: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct OrderResponse {
    pub order_id: String,
    pub total_price: f64,
    pub discount_applied: bool,
    pub final_price: f64,
}

File: Python/models/order.py
from typing import Optional, List
from pydantic import BaseModel

class OrderItem(BaseModel):
    item_id: str
    quantity: int

class OrderRequest(BaseModel):
    items: List[OrderItem]
    coupon_code: Optional[str]

class OrderResponse(BaseModel):
    order_id: str
    total_price: float
    discount_applied: bool
    final_price: float